src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -24,6 +81,19 @@ def __init__(self, path):
         self._parse()
         self.is_valid
 
+    def __lt__(self, other):
+        '''
+        Implementing this method and __eq__ allows the @total_ordering
+        decorator to turn the Device class into a totally ordered type.
+        This can slower then implementing all comparison operations.
+        This sorting should put valid devices before invalid devices and sort
+        on the path otherwise (str sorting).
+        '''
+        return (self._valid and not other._valid) or self.path < other.path
```

Comment:
This doesn't quite work - it needs to only do path ordering if both devices are valid, or both devices are invalid.

```suggestion
        return (self._valid and not other._valid) or (self._valid == other._valid and self.path < other.path)
```

Alternately, if that one-liner is too long or otherwise non-obvious:

```suggestion
        if self._valid == other._valid:
            return self.path < other.path
        return self._valid and not other._valid
```

Reply:
right ```invalid > valid``` should also be True. Thx, fixed and test case added.